From 482ad6c36d118aebdc063a2d7ffebbd16607d2c1 Mon Sep 17 00:00:00 2001
From: Adam Wolf <adamwolf@feelslikeburning.com>
Date: Tue, 12 Jun 2018 21:35:42 -0500
Subject: [PATCH] Handle STEP export properly on MacOS when launched from
 standalone pcbnew.
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="------------2.15.1"

This is a multi-part message in MIME format.
--------------2.15.1
Content-Type: text/plain; charset=UTF-8; format=fixed
Content-Transfer-Encoding: 8bit

---
 pcbnew/dialogs/dialog_export_step.cpp | 11 +++++++++++
 pcbnew/pcb_edit_frame.cpp             |  7 ++++++-
 2 files changed, 17 insertions(+), 1 deletion(-)


--------------2.15.1
Content-Type: text/x-patch; name="0001-Handle-STEP-export-properly-on-MacOS-when-launched-f.patch"
Content-Transfer-Encoding: 8bit
Content-Disposition: attachment; filename="0001-Handle-STEP-export-properly-on-MacOS-when-launched-f.patch"

diff --git a/pcbnew/dialogs/dialog_export_step.cpp b/pcbnew/dialogs/dialog_export_step.cpp
index 492d4c9b1..50d712bd4 100644
--- a/pcbnew/dialogs/dialog_export_step.cpp
+++ b/pcbnew/dialogs/dialog_export_step.cpp
@@ -249,6 +249,17 @@ void DIALOG_EXPORT_STEP::onExportButton( wxCommandEvent& aEvent )
     double yOrg = 0.0;
 
     wxFileName appK2S( wxStandardPaths::Get().GetExecutablePath() );
+    #ifdef __WXMAC__
+    // On macOS, we have standalone applications inside the main bundle, so we handle that here:
+        if( appK2S.GetPath().find( _("/Contents/Applications/pcbnew.app/Contents/MacOS") ) != wxNOT_FOUND )
+        {
+            appK2S.AppendDir( _(".."));
+            appK2S.AppendDir( _(".."));
+            appK2S.AppendDir( _(".."));
+            appK2S.AppendDir( _(".."));
+            appK2S.AppendDir( _("MacOS"));
+        }
+    #endif
     appK2S.SetName( "kicad2step" );
 
     wxString cmdK2S = "\"";
diff --git a/pcbnew/pcb_edit_frame.cpp b/pcbnew/pcb_edit_frame.cpp
index 5f9e930c5..938ac9a64 100644
--- a/pcbnew/pcb_edit_frame.cpp
+++ b/pcbnew/pcb_edit_frame.cpp
@@ -518,7 +518,12 @@ PCB_EDIT_FRAME::PCB_EDIT_FRAME( KIWAY* aKiway, wxWindow* aParent ) :
     // disable Export STEP item if kicad2step does not exist
     wxString strK2S = Pgm().GetExecutablePath();
     #ifdef __WXMAC__
-        strK2S += "Contents/MacOS/";
+    if (strK2S.find( _("pcbnew.app") ) != wxNOT_FOUND )
+    {
+        // On macOS, we have standalone applications inside the main bundle, so we handle that here:
+        strK2S += "../../";
+    }
+    strK2S += "Contents/MacOS/";
     #endif
     wxFileName appK2S( strK2S, "kicad2step" );
 

--------------2.15.1--


